### DISCUSSION TITLE
How can I get http status from a REST query?

### DISCUSSION THREAD

**Discussion by juangbm** [Budibase/budibase#9863] [Help] [answered]
Is it possible to access the HTTP status returned from a REST query?

![image](https://user-images.githubusercontent.com/22911293/222454403-18477696-b81b-4f2e-b550-4847af364660.png)

I am running the query inside a SAVE action, and I need to know if I can continue saving (if I get the correct result from the REST API,  status: 200) or stop saving and alert the user if something went wrong  

Thanks

**Comment by melohagan** [comment]
Hey @juangbm 

I don't believe there's a way to get the status code from the response header, however in your case I can see that the payload returns the status code.

You could use that in combination with [Action referencing](https://docs.budibase.com/docs/chaining-actions#action-referencing)  and the [Stop if action](https://docs.budibase.com/docs/chaining-actions#continue-if--stop-if)

**Comment by juangbm** [comment]
Only if it fails… and it is not exactly the status code… I will need to change the response to return always a result code in the response.

Thanks!


Juan González-Baylín Monge

Chief Product & Technology Officer

(+34) 626 55 19 90

www.dcoupon.es<http://www.dcoupon.es/>

www.scanbuyretail.com<http://www.scanbuyretail.com/>

Este mensaje va dirigido, de manera exclusiva, a su destinatario y puede contener información confidencial y sujeta al secreto profesional, cuya divulgación no está permitida por la ley. En caso de haber recibido este mensaje por error, le rogamos que, de forma inmediata, nos lo comunique mediante correo electrónico remitido a nuestra atención y proceda a su eliminación, así como a la de cualquier documento adjunto al mismo. Asimismo, le comunicamos que la distribución, copia o utilización de este mensaje, o de cualquier documento adjunto al mismo, cualquiera que fuera su finalidad, están prohibidas por la ley. En aras del cumplimiento de la LOPD y el RDLOP, puede ejercer los derechos A.R.C.O. de manera gratuita mediante email a ***@***.*** o en la dirección c/ José Castán Tobeñas 1, 28020 - Madrid.

This message is intended exclusively to its address and may contain confidential and subject to professional secrecy information whose disclosure is prohibited by law. If you have received this message in error, please, immediately, notify us via email sent to our care and to their elimination as well as any attachments thereto. We also inform that the distribution, copying or use of this message or any attachments thereto, whatever their purpose, are prohibited by law. In order to comply with the Data Protection Act, you can exercise your rights of access, rectification, cancellation or opposition for free via email to ***@***.***, or at the address c/ José Castán Tobeñas 1, 28020 - Madrid.

El 2 mar 2023, a las 19:18, melohagan ***@***.***> escribió:

﻿

Hey @juangbm<https://github.com/juangbm>

I don't believe there's a way to get the status code from the response header, however in your case I can see that the payload returns the status code.

You could use that in combination with Action referencing<https://docs.budibase.com/docs/chaining-actions#action-referencing> and [https://docs.budibase.com/docs/chaining-actions#continue-if--stop-if](Stop if action)

**Comment by melohagan** [comment]
Hey @juangbm 

I've had a look, and it appears there is no way to access the status code from an API response, unless I am mistaken.

Feature request here: https://github.com/Budibase/budibase/issues/9868

**Comment by AndresGarzonJ** [comment, marked as answer]
Hey @juangbm 

I have managed to access the HTTP response code via:

return helpers.JSONstringify($("Action 1.Query result").code);

"Action 1.Query result" comes from an "Execute Query".

**Reply by melohagan** [reply]
https://docs.budibase.com/docs/chaining-actions#query-result-properties
